src/Circulator.sol

```diff
@@ -187,14 +184,55 @@ contract Circulator is ICirculator, FeeOperator, Pausable, EIP712, Nonces {
             revert InvalidDelegateSignature();
         }
 
-        // Burn the token in TokenMessenger
-        nonce =
-            tokenMessenger.depositForBurn(burnAmt, delegateData.destinationDomain, delegateData.recipient, circleAsset);
+        nonce = _circulate(
+            permitData.sender,
+            delegateData.recipient,
+            permitData.amount,
+            delegateData.outputAmount,
+            fee,
+            delegateData.destinationDomain,
+            delegateData.fillDeadline,
+            delegateData.circulateType
+        );
+    }
+
+    function _circulate(
+        address _sender,
+        address _recipient,
+        uint256 _inputAmount,
+        uint256 _outputAmount,
+        uint256 _fee,
+        uint32 _destinationDomain,
+        uint32 _fillDeadline,
+        CirculateType _type
+    ) internal returns (uint64 nonce) {
+        uint256 burnAmount = _inputAmount - _fee;
+        if (burnAmount < _outputAmount) revert InsufficientBurnAmount();
```

Comment:
maybe rename this, `burnAmount` is only accurate for CCTP, so the error `InsufficientBurnAmount` makes less sense.

it can just be `finalAmount`, and rename error to `AmountAfterFeeLowerThanMinOutput` to be sepcific